### ISSUE TITLE
cargo build-bpf fails

### ISSUE THREAD

**Issue by aalayrot** [solana-labs/solana#22404]
#### Problem

When i try to run `cargo build-bpf`, i get the following error:

`BPF SDK: /Users/abdallaal-ayrot/.local/share/solana/install/releases/1.8.11/solana-release/bin/sdk/bpf
cargo-build-bpf child: rustup toolchain list -v
cargo-build-bpf child: cargo +bpf build --target bpfel-unknown-unknown --release
error: no such subcommand: '+bpf'
Did you mean 'b'?`

#### Environment
solana-cli 1.8.11 (src:423a4d65; feat:2385070269)
rustup 1.24.3 (ce5817a94 2021-05-31)
rustc 1.57.0 (f1edd0429 2021-11-29)
Macbook M1 11.2.3 (20D91)

**Comment by dmakarov** [comment]
BPF toolchain was not installed correctly on your system. You may want to try removing the cache
`rm -rf ~/.cache/solana` and run the `cargo build-bpf` command again. It will attempt to install the BPF toolchain afresh.

**Comment by dmakarov** [comment]
Also, what version of `cargo` do you have? `cargo -V`

**Comment by fabiooshiro** [comment]
The same here:
```
rm -rf ~/.cache/solana
cargo build-bpf
BPF SDK: /Users/oshiro/.local/share/solana/install/releases/1.9.9/solana-release/bin/sdk/bpf
cargo-build-bpf child: rustup toolchain list -v
cargo-build-bpf child: cargo +bpf build --target bpfel-unknown-unknown --release
error: no such subcommand: `+bpf`

	Did you mean `b`?
```
cargo 1.57.0
M1

I uninstall rust:
`brew uninstall rust`
And installed again and now it works

**Comment by fabiooshiro** [comment]
my problem was with:
`anchor build`
also I reinstalled `anchor` with:
`cargo install --git https://github.com/project-serum/anchor --tag v0.20.1 anchor-cli --locked`
not with `avm` as the first time

**Comment by rohitas-git** [comment]
I have an issue that I reinstall solana and thus, rust which was linked with prev solana files has changed. Previously I could use cargo build-bpf but now, I am shown error : 

BPF SDK path does not exist: /Users/user_name/.cargo/bin/sdk/bpf: No such file or directory (os error 2)

What should I do ?
rustup self uninstall is not working for me

**Comment by m4xw3llx** [comment]
```bash
BPF SDK path does not exist: /home/xxxx/.local/share/solana/install/releases/stable-3621da165accf04ad1805adfxxxx5c5dc9/solana-release/bin/sdk/bpf: No such file or directory (os error 2)
```
This problem occurs in my WSL2(Debian) when I execute `anchor test`. I tried to reinstall rustup and reinstall anchor,  they are not working. At last I've resolved it by remove solana and reinstall it.

```bash
rm -rf .local/share/solana
rm -rf .config/solana
rm -rf .cache/solana
sh -c "$(curl -sSfL https://release.solana.com/stable/install)"
```

**Comment by dmakarov** [comment]
```
export BPF_SDK_PATH=~/.local/share/solana/install/active_release/bin/sdk/bpf
```

**Comment by dmakarov** [comment]
reopen if this issue still exits with more recent solana version.

**Comment by adrianperreault** [comment]
This fixed it for me (on an M1 macOS 13.0) :
```
brew uninstall rust
curl https://sh.rustup.rs -sSf | sh
```

**Comment by crazydevlegend** [comment]
I'm trying to run the anchor demo project created by `anchor init` on M1 macOS 13.1.
I tried to build the project right after creating it, but it says `error: no such command: build-bpf`.

Currently, I'm using
```
- solana: 1.13.5
- rust, cargo: 1.25.2
- avm, anchor: 0.26.0
```

Can you please help me resolve this issue?
